app/helpers/stacks_helper.rb

```diff
@@ -26,4 +38,24 @@ def pull_request_link(commit)
   def render_raw_commit_id_link(commit)
     link_to(commit.short_sha, github_commit_url(commit), target: '_blank', class: 'number')
   end
+
+  private
+
+  def deploy_button_disabled?(commit)
+    !commit.deployable? || commit.stack.locked?
+  end
+
+  def deploy_button_caption(commit)
+    return 'Locked' if commit.stack.locked?
+
+    if commit.deployable?
+      return 'Deploy in progress...' if commit.stack.deploying?
+      return 'Deploy'
+    end
+
+    return 'CI Pending...' if commit.pending?
+    return 'CI Failure' if commit.failure?
+    return 'CI Error' if commit.error?
+    'Not Run'
+  end
```

Comment:
Sorry I fixed it, but look like I forgot to commit the change. I just pushed it on master.